src/services/llms/base.py

```diff
@@ -1,13 +1,79 @@
+# src/services/llm/base.py
 from abc import ABC, abstractmethod
-from typing import List, Dict, Any
+import json
+from typing import List, Dict
+from unidiff import Hunk, PatchedFile
 from ...core.models import PRDetails
 
-
 class BaseLLMService(ABC):
+    """
+    Abstract base class for LLM services defining the common interface
+    that all LLM implementations must follow.
+    """
+    
     @abstractmethod
-    def create_prompt(self, file_path: str, diff_content: str, pr_details: PRDetails) -> str:
```

Comment:
The method name `create_prompt` is not very descriptive.  Consider renaming it to something like `generate_prompt` or `construct_prompt` for better clarity.  This will improve the readability and understanding of the code.